node/subsystem-types/src/errors.rs

```diff
@@ -75,13 +75,17 @@ pub enum RecoveryError {
 
 	/// A requested chunk is unavailable.
 	Unavailable,
+
+	/// Internal error, channel closed.
+	Internal,
```

Comment:
Does this mean node is shutting down?

Reply:
Technically, that's when it happens, otherwise it might just be any other issue that brings the subsystem down.

Reply:
Maybe it is good to add more variants for this error. It took me some time to understand the logic, and the name `Internal` is really not very helpful.

Reply:
I've renamed it to ChannelClosed as it applies to all occurences.